### PR TITLE
mspec updates through 202411051711

### PR DESCRIPTION
**Pull request by headius**

### CODE REVIEW

**Comment by headius** [comment]
@eregon Please review. The `wordsize` logic could also just be left in place, aliased to `c_long_size` but I chose to deprecate it in deference to CRuby.

File: lib/mspec/helpers/numeric.rb [hunk 1]
```diff
@@ -15,11 +16,13 @@ def bignum_value(plus = 0)
 end
 
 def max_long
-  2**(0.size * 8 - 1) - 1
+  long_byte_size = [0].pack('l!').size
```
Comment (eregon) [inline lib/mspec/helpers/numeric.rb:19]: We should use `PlatformGuard::C_LONG_SIZE` here and below to avoid recomputing it every single time.

**Review by eregon — APPROVED** [review]
Thank you, this looks good, and I think it's worth keeping the word_size stuff a bit longer, deprecating it should be enough to avoid having new usages.

File: lib/mspec/helpers/numeric.rb (hunk 1 above)
Comment (eregon) [reply lib/mspec/helpers/numeric.rb:19]: I'll do it

Comment (eregon) [reply lib/mspec/helpers/numeric.rb:19]: https://github.com/ruby/mspec/commit/6499325c5fc4044f26a41ee3a769a67d79062229

File: lib/mspec/guards/platform.rb [hunk 2]
```diff
@@ -54,22 +54,34 @@ def self.wsl?
...
 
   WORD_SIZE = 1.size * 8
+  deprecate_constant :WORD_SIZE
 
   POINTER_SIZE = begin
     require 'rbconfig/sizeof'
     RbConfig::SIZEOF["void*"] * 8
   rescue LoadError
-    WORD_SIZE
+    [0].pack('j').size
```
Comment (eregon) [inline lib/mspec/guards/platform.rb:63]: This fallback and the one below were incorrect, fixed in https://github.com/ruby/mspec/commit/92ee9f7c8cdeb72f3b58c755cf671e2a27227e02

Comment (headius) [reply lib/mspec/guards/platform.rb:63]: Ah good catch.

File: lib/mspec/helpers/numeric.rb (hunk 1 above)
Comment (headius) [reply lib/mspec/helpers/numeric.rb:19]: Good catch.

**Pull request merged**